comms/src/protocol/messaging/protocol.rs

```diff
@@ -195,9 +195,19 @@ impl MessagingProtocol {
         trace!(target: LOG_TARGET, "Internal messaging event '{}'", event);
         match event {
             OutboundProtocolExited(node_id) => {
-                self.active_queues.remove(&node_id).expect(
-                    "OutboundProtocolExited event, but MessagingProtocol has no record of the outbound protocol",
+                debug!(
+                    target: LOG_TARGET,
+                    "Outbound protocol handler exited for peer `{}`",
+                    node_id.short_str()
                 );
+                if self.active_queues.remove(&node_id).is_none() {
+                    debug!(
```

Comment:
nit: Maybe a warn here?

Reply:
I don't think so - it's actually something that we'd expect from time to time because there is a race condition here. However the purpose of this is just to clean up a dead node ID / channel pair - so either way, the race condition doesn't affect that goal.